packages/pigeon/lib/cpp_generator.dart

```diff
@@ -631,19 +627,19 @@ const flutter::StandardMessageCodec& ${api.name}::GetCodec() {
                     elseBody = '''
 $prefix\tauto output_optional = $extractedValue;
 $prefix\tif (output_optional) {
-$prefix\t\twrapped.emplace($resultKey, $wrapperType(std::move(output_optional).value()));
+$prefix\t\twrapped.push_back($wrapperType(std::move(output_optional).value()));
 $prefix\t} else {
-$prefix\t\twrapped.emplace($resultKey, $nullValue);
+$prefix\t\twrapped.push_back($nullValue);
 $prefix\t}${indent.newline}''';
                   } else {
                     elseBody =
-                        '$prefix\twrapped.emplace($resultKey, $wrapperType($extractedValue));${indent.newline}';
+                        '$prefix\twrapped.push_back($wrapperType($extractedValue));${indent.newline}';
                   }
                   ifCondition = 'output.has_error()';
                   errorGetter = 'error';
                 }
                 return '${prefix}if ($ifCondition) {${indent.newline}'
-                    '$prefix\twrapped.emplace($errorKey, WrapError(output.$errorGetter()));${indent.newline}'
+                    '$prefix\twrapped = WrapError(output.$errorGetter());${indent.newline}'
                     '$prefix$reply'
```

Comment:
For future generated code streamlining: I was looking at the generated code to review changes and noticed that we're producing:
```
if (output.has_value()) {
  wrapped = WrapError(output.value());
  reply(wrapped);
} else {
  wrapped.push_back(flutter::EncodableValue());
  reply(wrapped);
}
```

This would be a little shorter and cleaner as:

```
if (output.has_value()) {
  wrapped = WrapError(output.value());
} else {
  wrapped.push_back(flutter::EncodableValue());
}
reply(wrapped);
```